app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -256,7 +284,31 @@ class ShowAlternativeRoutesActivity : AppCompatActivity() {
                         .build()
                 )
                 .build()
-        )
+        ).attach(this)
+
+        lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                MapboxNavigationApp.registerObserver(navigationCreatedObserver)
+            }
+
+            override fun onResume(owner: LifecycleOwner) {
+                MapboxNavigationApp.registerObserver(navigationResumedObserver)
```

Comment:
If I remember correctly, we used to register/unregister observers in onStart/onStop, not onResume/onPause. Is this change expected?

Reply:
You get to choose where to register/unregister, and each lifecycle pair has reasons to choose it.

For example, register/unregister observers that will inflate views in on **created/destroyed**. That way you do not re-create views every time the app is foregrounded.

Alternatively, register/unregister observers that share data in **start/stop**. That way data is saved between fragment and activity transitions.

Also, register/unregister observers that will poll on networks to update views in **resume/pause**. That way you're not requesting network data while the app is backgrounded.